### PR TITLE
Make AppendRuntimeIdentifierToPath=true the default

### LINKED ISSUE #868: RID-specific builds should use RID-specific Output Path
We have identified a regression between project.json and csproj which is preventing users from generating deterministic build outputs.

Because this issue prevents us from providing deterministic build outputs, we will need to fix it at some point. I’d rather fix it in 1.0 RTM, before the change becomes `breaking relative to 1.0`.
 
### What we had in Project.json
In `project.json` the CLI would append the RID to a project’s output path if the output contained RID-specific artifacts. This meant that:
 
```
Portable Project
  These would output to:
 
    <projdir>/bin/Debug/{TFM}/
 
  RID-Specific project
  These would output to:
     <projdir>/bin/Debug/{TFM}/{RID}/
 ```

### What we have in csproj
```
  Portable Project
  These would output to:
 
    <projdir>/bin/Debug/{TFM}/
 
  RID-Specific project
  These would output to:
     <projdir>/bin/Debug/{TFM}/
 ```
 
## What’s the issue?
In the new SDK we do not have a unique output path for common build configurations. Consider these scenarios:
 
- Portable Project
  - dotnet build
- Multi-TFM project
  - dotnet build –framework=netcoreapp1.0
  - dotnet build –framework=netcoreapp1.1
- Multi-RID project
  - dotnet build –runtime=win7-x64
  - dotnet build –runtime=win7-x86
 
In the first two project types, everything is fine. The portable project has but a single output. The multi-TFM project has two outputs, but each goes into a TMF-specific directory.
 
The Multi-RID project is the problem. Both of those `build` invocations will result in outputs being generated into the same directory, despite the outputs being meaningfully different!
 
What is worse, this error will likely occur silently. Our own incrementality checks will assume that the x86 build produced valid outputs for x64 and will simply return.
 
## How to fix?
We already have a property, `AppendRuntimeIdentifierToOutputPath`, that is defaulted to ‘false’. We would like, instead, to set this property to `true` by default IFF the current build has a `RuntimeIdentifier` specified. This will provide a unique output path for every build gesture advertised to users. It will also simplify fixes for several `race condition` style bugs that we’re facing because RIDs do not have unique output paths.
 
## What’s the risk?
Early adopters of the SDK [anyone using it so far] may have dependencies on the output path of their RID-specific builds. They will have two recourses once this change is made:
a. Update their dependencies to use the new output path
b. Set AppendRuntimeIdentifierToOutputPath=false in their csproj/on the cmd line
 

/cc @srivatsn @MattGertz @nguerrera @dsplaisted

### PR DESCRIPTION
**Pull request by nguerrera**
Fix #868

@dsplaisted @srivatsn @eerhardt @piotrpMSFT

### CODE REVIEW

**Review by srivatsn — APPROVED** [review]

File: test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildASelfContainedApp.cs [hunk 1]
```diff
@@ -32,8 +32,6 @@ public void It_builds_a_runnable_output()
                     var ns = project.Root.Name.Namespace;
                     var propertyGroup = project.Root.Elements(ns + "PropertyGroup").First();
                     propertyGroup.Add(new XElement(ns + "RuntimeIdentifier", runtimeIdentifier));
-
-                    propertyGroup.Add(new XElement(ns + "AppendRuntimeIdentifierToOutputPath", "true"));
```
Comment (eerhardt) [inline test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildASelfContainedApp.cs]: Should we maybe set this to "false" to test out that the RID doesn't get appended to the output path?

**Review by eerhardt — APPROVED** [review]

**Review by dsplaisted — APPROVED** [review]

**Comment by srivatsn** [comment]
@MattGertz for approval.

**Comment by srivatsn** [comment]
Just noticed there's no template. Hold on.

**Comment by srivatsn** [comment]
**Customer scenario**

Builds that build with multiple rids can stomp their output on each other because the outputpath is not RID specific.

**Bugs this fixes:** 

https://github.com/dotnet/sdk/issues/868

**Workarounds, if any**
Set AppendRuntimeIdentifierToPath=true manually in the project.

**Risk**

Medium - since we are changing output paths.

**Performance impact**

Low - we are just choosing a different output path.

**Is this a regression from a previous update?**

No.

**Root cause analysis:**

Originally we tried making RIDs a publsh-only concept but that didn`t work out. This is a holdover from that time.

**How was the bug found?**

Internal testing of other RID changes.

**Comment by MattGertz** [comment]
Yep, I mentioned this one to JoC earlier, let's take it to Shiproom.

**Comment by nguerrera** [comment]
@dotnet-bot Test Ubuntu14.04 Release please

**Comment by nguerrera** [comment]
@dotnet-bot Test Ubuntu14.04 Debug please

**Comment by nguerrera** [comment]
Retests are for myget connectivity issue

**Comment by nguerrera** [comment]
@dotnet-bot Test Ubuntu14.04 Release please

**Comment by nguerrera** [comment]
@dotnet-bot Test Ubuntu14.04 Debug please

**Comment by nguerrera** [comment]
@dotnet-bot Test Windows_NT Release please

**Comment by nguerrera** [comment]
@dotnet-bot Test Windows_NT Debug please

File: test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildASelfContainedApp.cs (hunk 1 above)
Comment (nguerrera) [inline test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildASelfContainedApp.cs]: Coverage for on and off went in with #847

**Review by livarcocc — APPROVED** [review]

**Comment by nguerrera** [comment]
@dotnet-bot test this please (myget should be back up now)

**Comment by srivatsn** [comment]
VS Mac and ASP NET folks signed off on this change. Merging.

**Comment by natidea** [comment]
We will bring a new insertion PR to shiproom today for approval to get this one.

**Pull request merged**